tests/test_RadClass.py

```diff
@@ -91,14 +86,52 @@ def test_cache():
 
     # run handler script
     classifier = RadClass(stride, integration, test_data.datapath,
-                          test_data.filename, store_data=True,
+                          test_data.filename,
                           cache_size=cache_size)
     classifier.run_all()
 
     # the resulting 1-hour observation should be:
     #   counts * integration / live-time
     expected = spectra * integration / test_data.livetime
-    results = np.genfromtxt('results.csv', delimiter=',')[1, 1:]
+    results = classifier.storage.to_numpy()[1]
+    np.testing.assert_almost_equal(results, expected[0], decimal=2)
+
+
+def test_stride():
+    stride = 100
+    integration = 50
+
+    # run handler script
+    classifier = RadClass(stride, integration, test_data.datapath,
+                          test_data.filename)
+    classifier.run_all()
+
+    # the resulting 1-hour observation should be:
+    #   counts * integration / live-time
+    expected = spectra * integration / test_data.livetime
+    expected_samples = int(test_data.timesteps /
+                           (integration + (stride - integration)))
+    np.testing.assert_almost_equal(classifier.storage.iloc[0],
```

Comment:
Isn't it important to tests the 2nd result to make sure the stride worked?  The first result will be the same independent of the stride.

Reply:
I have changed it to test the second result. I should note that because the `spectra` matrix is completely identical (i.e. each `[ij]` is 10.0), the value will be the same for each result. The first `assert` is just checking to make sure the integration math is still correct when stride != integration (similar to `test_integration`) while the second `assert` ensures the expected number of results when stride != integration. Perhaps it would be more interesting to have a `spectra` matrix that is not identical, but I'm not sure how to construct tests for a random matrix...

Reply:
I agree that a non-uniform spectra matrix is a good idea

Reply:
Since `spectra` is global here, would it be sufficient enough to manually sum the correct indices in the test (using something like `np.sum()`) and compare that to what happens in `RadClass` or would that be too similar to `RadClass`'s own operations? This would have to be changed for several of the tests in `test_RadClass`.

Reply:
For a known spectra you could simply pre-calculate the correct answer to the sum.

e.g. `spectra[i] = i` then 
`sum(spectra,j,k) = [(k+1)k - j(j-1)]/2` (or something like that)